src/Explorer/Panes/SettingsPaneR/index.tsx

```diff
@@ -0,0 +1,433 @@
+import { PrimaryButton } from "office-ui-fabric-react";
+import React, { FunctionComponent, KeyboardEvent, useEffect, useLayoutEffect, useRef, useState } from "react";
+import * as Constants from "../../../Common/Constants";
+import { configContext } from "../../../ConfigContext";
+import { LocalStorageUtility, StorageKey } from "../../../Shared/StorageUtility";
+import * as StringUtility from "../../../Shared/StringUtility";
+import * as NotificationConsoleUtils from "../../../Utils/NotificationConsoleUtils";
+import Explorer from "../../Explorer";
+
+export interface SettingsPaneProps {
+  explorer: Explorer;
+  closePanel: () => void;
+}
+
+export const SettingsPaneR: FunctionComponent<SettingsPaneProps> = ({ explorer, closePanel }: SettingsPaneProps) => {
+  const [formErrors, setFormErrors] = useState("");
+  const [formErrorsDetails, setFormErrorsDetails] = useState();
+  const [isExecuting, setIsExecuting] = useState(false);
+  const [pageOption, setPageOption] = useState("");
+  const [customItemPerPage, setCustomItemPerPage] = useState(0);
+  const [crossPartitionQueryEnabled, setCrossPartitionQueryEnabled] = useState(
+    LocalStorageUtility.hasItem(StorageKey.IsCrossPartitionQueryEnabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.IsCrossPartitionQueryEnabled) === "true"
+      : true
+  );
+  const [graphAutoVizDisabled, setGraphAutoVizDisabled] = useState(
+    LocalStorageUtility.hasItem(StorageKey.IsGraphAutoVizDisabled)
+      ? LocalStorageUtility.getEntryString(StorageKey.IsGraphAutoVizDisabled)
+      : "false"
+  );
+  const [maxDegreeOfParallelism, setMaxDegreeOfParallelism] = useState(
+    LocalStorageUtility.hasItem(StorageKey.MaxDegreeOfParellism)
+      ? LocalStorageUtility.getEntryNumber(StorageKey.MaxDegreeOfParellism)
+      : Constants.Queries.DefaultMaxDegreeOfParallelism
+  );
```

Comment:
Add <type> parameters for all useState hooks

Reply:
done